electrumsv/tests/test_wallet.py

```diff
@@ -486,128 +510,294 @@ def test_multisig(self, mock_app_state, tmp_storage) -> None:
-    with unittest.mock.patch(
-            'electrumsv.keystore.app_state.credentials.add_indefinite_credential',
-            add_indefinite_credential_mock):
-        try:
-            wallet = Wallet(storage)
-        except FileNotFoundError as e:
-            if sys.version_info[:3] >= (3, 8, 0):
-                msg = "Could not find module 'libusb-1.0.dll' (or one of its dependencies)."
-                if msg in e.args[0]:
-                    pytest.xfail("libusb DLL could not be found")
-                    return
-            raise e
-        except OSError as e:
-            if sys.version_info[:3] < (3, 8, 0):
-                if "The specified module could not be found" in e.args[1]:
-                    pytest.xfail("libusb DLL could not be found")
-                    return
-            raise e
-
-    # Store any pre-password update related data to compare against post-password update data.
-    prv_keypairs: dict[str, str] = {}
-    if "imported" == expected_type and "privkey" in wallet_filename:
-        assert len(wallet.get_accounts()) == 2
-        private_key_account = cast(ImportedPrivkeyAccount,
-            [ entry for entry in wallet.get_accounts() if not entry.is_petty_cash() ][0])
-        private_key_keystore = cast(Imported_KeyStore, private_key_account.get_keystore())
-        # Pre-decrypt the prv for later comparison so the initial password is not needed there.
-        for public_key, encrypted_prv in private_key_keystore._keypairs.items():
-            prv_keypairs[public_key.to_hex()] = pw_decode(encrypted_prv, initial_password)
-
-    password = "654321"
-    future, update_completion_event = wallet.update_password(initial_password, password)
-    # Wait for the database update to finish.
-    future.result(5)
-    # Wait for the done callback to finish.
-    update_completion_event.wait()
-
-    if "standard" == expected_type:
-        check_legacy_parent_of_standard_wallet(wallet, password=password,
-            add_indefinite_credential_mock=add_indefinite_credential_mock)
-    elif "imported" == expected_type:
-        if "privkey" in wallet_filename:
-            check_legacy_parent_of_imported_privkey_wallet(wallet, password, prv_keypairs)
-        elif "address" in expected_subtypes:
-            check_legacy_parent_of_imported_address_wallet(wallet)
+    # Derivation data in the MasterKeys table.
+    count_coverage_masterkeys_table_seed: int
+    count_coverage_masterkeys_table_passphrase: int
+    count_coverage_masterkeys_table_xprv: int
+
+    # Newer wallets have a `Servers` table with an `encrypted_api_key` column
+    count_coverage_server_api_key: int
+
+    # Note that while the CredentialCache does contain sensitive data, it is not encrypted and
+    # therefore does not need to be checked for re-encryption coverage.
+
+
+class TestLegacyWalletLoading:
+
+    test_state = TestState(
+        failing_wallets=set(),
+        count_coverage_password_token=0,
```

Comment:
Put default values in the dataclass `count_blah; int = 0`, remove all instantiation values.